modules/pipeline/pkg/containers/task_container.go

```diff
@@ -15,25 +15,22 @@
 package containers
 
 import (
-	"encoding/json"
 	"fmt"
 
 	"github.com/erda-project/erda/apistructs"
 	"github.com/erda-project/erda/modules/pipeline/spec"
 )
 
 func GenContainers(task *spec.PipelineTask) ([]apistructs.TaskContainer, error) {
-	if value, ok := task.Extra.Action.Params["bigDataConf"]; ok {
-		spec := apistructs.BigdataSpec{}
-		if err := json.Unmarshal([]byte(value.(string)), &spec); err != nil {
-			return nil, fmt.Errorf("failed to unmarshal task bigDataConf")
-		}
+	if spec, err := task.GetBigDataConf(); err == nil {
```

Comment:
bad pattern. you split the two normal situations (flink+spark and task).
return as soon as possible if err occurred.
then check spec fields.

Reply:
adjusted